hack/test-cmd.sh

```diff
@@ -666,7 +666,7 @@ runTests() {
   # Command: autoscale rc "frontend"
   kubectl autoscale -f examples/guestbook/frontend-controller.yaml --save-config "${kube_flags[@]}" --max=2
   # Post-Condition: hpa "frontend" has configuration annotation
-  [[ "$(kubectl get hpa frontend -o yaml "${kube_flags[@]}" | grep kubectl.kubernetes.io/last-applied-configuration)" ]]
+  [[ "$(kubectl get hpa.extensions frontend -o yaml "${kube_flags[@]}" | grep kubectl.kubernetes.io/last-applied-configuration)" ]]
```

Comment:
> Is HPA only in extensions now?

You can see in the line below that `hpa` is an unambiguous resource.  This line demonstrates that it *can* be disambiguated, not that it must.

The `PriorityRESTMapper` pr doesn't force this to be fully qualified either.  It simply says: "given an ambiguous resource match, do I have a rule that would prioritize the one match that I should use".  The combination of these pulls allows you to have `kubectl get pods` use `"",v1,pods` instead of `metrics,v1beta1,pods` without any qualification and this pull lets you sayd `kubectl get pods.metrics` to get the resource you want.